mlir/include/mlir/Dialect/Linalg/TransformOps/LinalgTransformOps.td

```diff
@@ -390,6 +390,42 @@ def GeneralizeOp : Op<Transform_Dialect, "structured.generalize",
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// SpecializeOp
+//===----------------------------------------------------------------------===//
+
+def SpecializeOp : Op<Transform_Dialect, "structured.specialize",
+    [FunctionalStyleTransformOpTrait, MemoryEffectsOpInterface,
+     TransformOpInterface, TransformEachOpTrait,
+     ReportTrackingListenerFailuresOpTrait]> {
+  let description = [{
+    Transforms a generic operation into the equivalent named form.
+
+    #### Return modes
+
+    This operation ignores non-Linalg ops and drops them in the return.
+    If all the operations referred to by the `target` handle specialize
+    properly, the transform succeeds. Otherwise the transform silently fails.
```

Comment:
Nit: it's probably my fault in naming, but please don't use "silently fails". The failure is not silent, it's _silenceable_. It _will be reported_ by default unless the surrounding context requests that otherwise.

Reply:
Thanks, updated!